feat: implement Wheel of Destiny

vllsystems opened the pull request:
# Description

Implement the Wheel of Destiny system for OTClient.

## Behavior

### **Actual**

None

### **Expected**

Players can open the Wheel of Destiny interface, view their skill specializations, and interact with the wheel nodes to allocate points.  

## Fixes

\# (issue)

## Type of change

Please delete options that are not relevant.

 - [ ] Wheel of Destiny graphical interface  
 - [ ] Server data parsing (ResourceTypes.WHEEL_OF_DESTINY)  
 - [ ] Specialized skills and points system  
 - [ ] Integration with existing skills system  
 - [ ] Informative tooltips for each wheel node  
 - [ ] Visual animations and effects  

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

  - [ ] Test A
  - [ ] Test B

**Test Configuration**:

  - Server Version: Canary Lastest
  - Client: OTCR Lastest
  - Operating System: Windows 10

## Checklist

  - [ ] My code follows the style guidelines of this project
  - [ ] I have performed a self-review of my own code
  - [ ] I checked the PR checks reports
  - [ ] I have commented my code, particularly in hard-to-understand areas
  - [ ] I have made corresponding changes to the documentation
  - [ ] My changes generate no new warnings
  - [ ] I have added tests that prove my fix is effective or that my feature works

vllsystems committed:
wheel

vllsystems committed:
constants and configurations

vllsystems committed:
images

kokekanon commented:
Do you need help? About six months ago, I did the cpp part(protocolparse). Do you want it?

vllsystems commented:
> Do you need help? About six months ago, I did the cpp part(protocolparse). Do you want it?

ty @kokekanon 
I'm not rushing; I'm doing it in my own time.

vllsystems committed:
update image patch and fix image clip

vllsystems committed:
17-10

vllsystems committed:
18-10

vllsystems committed:
19-10

vllsystems commented:
A kind of roadmap would be good to guide the development. I'm still trying to understand some workings of the wheel before applying the nodes.

gabrielew committed:
wip: top left quadrant

gabrielew committed:
feat: top_left

gabrielew committed:
Merge pull request #115 from gabrielew/feat/wheel-top-left

Feat/wheel top left

gabrielew committed:
wip: add values to another quadrants

gabrielew committed:
feat: top right

gabrielew committed:
feat: reset slices

gabrielew committed:
feat: bottom left and right quadrants

gabrielew committed:
feat: reset bottom quadrants

vllsystems committed:
Merge pull request #1 from gabrielew/feat/wheel_of_destiny

Feat/wheel of destiny

vllsystems committed:
olha a fera ai

vllsystems committed:
implement dynamic slot progress loading and click handlers

Refactor quadrant progress data to wod_information.lua
Add slot click handlers with border toggle functionality
Implement dynamic image updates based on progress frames
Load vocation-specific progress data with fallback to defaults
Fix slot widget hierarchy and event handling

vllsystems commented:
> Do you need help? About six months ago, I did the cpp part(protocolparse). Do you want it?


https://github.com/user-attachments/assets/9417257c-25e7-4257-876f-f7443e692697

Greetings, Master Kokekanon.

Yes, we are going to need help. I believe it is a clip-image problem, as you can see in the video. The bottomLeft's node 5 is recognizing the 9, even though it is correct in Lua and the rest of the OTUI. Also, some nodes are "larger" than they should be.

Any idea how to solve this? I don't really know how clip-image works.

libergod commented:
Hello @vllsystems can we close this PR ? So that I can continue development under https://github.com/mehah/otclient/pull/1398 

Thanks for the response.
